File: 프로그래머스/lv0/120812. 최빈값 구하기/최빈값 구하기.js
function solution(array) {
    var value = [0];
    var answer = 0;
    var cnt = {};
    for(e of array){
        if(cnt[e]==null) cnt[e] = 1;
        else cnt[e]++;
    }
    for(key in cnt){
        if(cnt[key] >= value[0]) {
            value.push(cnt[key]);
            value.sort((a,b) => b-a);
            answer = key;
        }
    }
    return value[0]>value[1] ? Number(answer) : -1;
}

File: 프로그래머스/lv0/120812. 최빈값 구하기/최빈값 구하기.java
class Solution {
    public int solution(int[] array) {
        int answer = 0, max = -1;
        int[] index = new int[1000];
        for(int arr: array){
            index[arr]++;
            if(index[arr]>max){
                max = index[arr];
                answer = arr;
            }else if(index[arr]==max) { answer = -1; }
        }
        return answer;
    }
}